taf/src/main/java/com/taf/automation/api/ApiDomainObject.java

```diff
@@ -198,6 +215,28 @@ public DataAliases getGlobalAliases() {
         return dataAliases;
     }
 
+    /**
+     * This method will get the token from the cache if it exists otherwise it will use client to get the token
+     *
+     * @param client - Client to get token if necessary and client will be updated with token
+     * @param key    - Key to get token.  If tokens are tied to roles, then the role should be used as the key,
+     *               otherwise pick something to make it unique
+     */
+    protected static void cacheToken(ApiLoginSession client, String key) {
+        String token = tokenCache.getIfPresent(key);
+        if (token == null) {
+            synchronized (ApiDomainObject.class) {
+                token = tokenCache.getIfPresent(key);
+                if (token == null) {
+                    token = client.getToken();
```

Comment:
This only gets the existing token which most likely would be null.  There needs to be a request to get a valid token first.  Since, this is just an example use the client.login method.